### PR TITLE
fix(hooks): extract Claude session_id via jq in sandbox (closes #1760)

### LINKED ISSUE #1760: Replace grep-based Claude session_id hook extraction with JSON-aware parsing
_Note: this issue was drafted by Claude via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

#1735 added a hook-driven sidecar that captures Claude's `session_id` by piping the hook's stdin JSON through a `grep` pipeline (`hook_command_session_id` in `src/hooks/mod.rs`), writing `/tmp/aoe-hooks/<id>/session_id`.

The grep approach carries a documented fragility:

- It relies on the top-level `session_id` appearing textually before any nested `session_id`. If Claude ever emits a payload where a nested copy precedes the top-level field, `head -1` captures the wrong UUID. `test_hook_command_session_id_locks_top_level_first_contract` pins the current behavior rather than asserting correctness.
- The regression tests use synthetic payloads, so a real Claude payload schema change would not be caught by CI; it would surface in the field as a wrong-conversation resume.

**Proposed remediation:** Replace the grep pipeline with a small `aoe` subcommand (for example `aoe hook capture-session-id`) invoked by the hook. It reads stdin, parses JSON with serde to pull the top-level `session_id`, and writes the sidecar atomically. This removes both the key-ordering dependency and the prompt-injection regex, and is testable against real payloads.

**Minor, same mechanism:** the sidecar lives at a predictable `/tmp/aoe-hooks/<id>/session_id`; on a multi-user host another local user could plant a value (low impact: resume-target only, validated as UUID plus 5-minute staleness). Could be hardened (0700 dir plus ownership check) in the same pass.

Follow-up to #1735.

### PR DESCRIPTION
**Pull request by jerome-benoit**
## Description

Replace the shell `grep`/`tr` pipeline in `hook_command_session_id_sandbox` with a `jq -r` filter doing structural top-level extraction, symmetric with the host variant's `serde_json` route in [`aoe __extract-session-id`](src/cli/extract_session_id.rs). The host variant was already JSON-aware via #1735; the sandbox variant was left with a fragile regex that depended on textual order.

The previous body used `[{,]\s*"session_id"\s*:\s*"<UUID>"` as the grep anchor and `head -1` to pick the textually-first match. That accepts both top-level field starts and nested-object first-field starts, so a payload like `{"context":{"session_id":NESTED},"session_id":TOP}` (nested before top in textual order) wrote NESTED. Test `test_hook_command_session_id_sandbox_pins_nested_first_quirk` documented the bug; this PR inverts it to assert TOP wins.

The new body uses the `jq` filter `if (.session_id|type)=="string" then .session_id else empty end`, deliberately core-only (no `test()`, so independent of how `jq` was built). The UUID 8-4-4-4-12 shape is enforced afterwards by a POSIX `case` over `H=[0-9a-fA-F]`. All existing shell guards stay in place: `unset IFS; set -f; umask 077`, the `AOE_INSTANCE_ID` allowlist, the mode-0700 dir check, atomic `.tmp` + `mv`, and the trailing `exit 0 # aoe-hooks` sentinel.

Closes #1760.

## Why the host variant is untouched

This PR's diff is scoped to `src/hooks/mod.rs`. The host variant ([`hook_command_session_id_host()`](src/hooks/mod.rs)) and [`src/cli/extract_session_id.rs`](src/cli/extract_session_id.rs) are byte-identical to `main`.

## Sandbox image impact

`jq` is already installed in the default `aoe-sandbox` image at [docker/Dockerfile:14](docker/Dockerfile). No Dockerfile change. On a custom sandbox image without `jq`, `command -v jq >/dev/null 2>&1 || exit 0` triggers silent-fail to exit 0 (no sidecar written), matching the host variant's silent-failure mode when `aoe` is off PATH; pane-content reconciliation takes over.

## Migration

None. Sandbox hook strings live in agent settings files under sandbox subpaths (e.g. `~/.claude/sandbox/settings.json`), which `v015` and `v017` explicitly skip. [`src/session/container_config.rs:1517-1546`](src/session/container_config.rs) re-stages these via `install_hooks(..., HookInstallTarget::Sandbox)` on every session creation, so the canonical bytes land at the next container start with no operator action.

## Out of scope

- Hardening the in-container `/tmp/aoe-hooks/<id>` path. Confirmed harmless: `read_hook_session_id` reads from `dir_guard::hook_base_path()` = `/tmp/aoe-hooks-<euid>` host-side, which is already per-user-hardened. The container path is the bind-mount destination of that hardened source, single-tenant per container. A cosmetic follow-up could bake the euid suffix into `HOOK_STATUS_BASE_IN_CONTAINER` for symmetry, but it carries no security weight.
- Host variant or `src/cli/extract_session_id.rs` changes.
- Other hooks (status, urgent/attention).
- The three pre-existing test flakes (#1913 reconcile, #2231 tmux × 2).

## PR Type

- [ ] New Feature
- [x] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## Checklist

- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] N/A: this PR doesn't change a user-facing dashboard flow
- [ ] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`
- [ ] Skipped a test, because:

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle
- [ ] Added or updated an e2e test under `tests/e2e/`
- [ ] Skipped a test, because:

The regression contract is locked at the unit level via `src/hooks/mod.rs`'s shell-emitter test suite, which spawns the live `sh -c` body against real payloads:

- `test_hook_command_session_id_sandbox_extracts_top_level_over_nested` (inverts the former `..._pins_nested_first_quirk`; locks the #1760 fix).
- `test_hook_command_session_id_sandbox_extracts_from_real_claude_payload` exercises a Claude `PreToolUse`-shape payload with a nested `tool_input.session_id` decoy and a literal `"session_id":"<uuid>"` substring inside the user `prompt` field; asserts the real top-level UUID wins.
- `test_hook_command_session_id_sandbox_invokes_jq_extractor` and the extended `hook_command_session_id_sandbox_quotes_and_guards` assert the shell tokens (`jq -r`, `command -v jq`, `H=[0-9a-fA-F]`, atomic `.session_id.$$.tmp`) and forbid the legacy `grep -oE`.
- All shell-driven sandbox tests are gated by `skip_if_no_jq()` so minimal CI runners without `jq` skip rather than spuriously fail.

The host-variant contract test `test_hook_command_session_id_host_invokes_aoe_subcommand` is untouched.

## How tested

```
cargo fmt --all -- --check    # clean
cargo clippy --features serve --all-targets -- -D warnings    # clean
cargo test --features serve --lib    # 4624 passed, 3 pre-existing flakes (#1913, #2231 x2) unrelated
```

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:** claude-opus-4-7 (Anthropic), orchestrated via OhMyOpenCode (Sisyphus agent + Oracle consultations).

**Any Additional AI Details you'd like to share:** Design choice between (A) bundling the `aoe` binary into the container, (B) `jq`/`python3` inline, (C) baking an extractor script at sandbox setup, (D) a depth-aware shell pipeline was selected through an Oracle consultation pass before implementation, and a second Oracle pass reviewed the final diff. Option B with `jq` (already present in the default image) was chosen for smallest correct change with no Dockerfile edit, no migration, and clean cross-arch story.

- [x] I am an AI Agent filling out this form (check box if true)

This PR changes how the sandbox hook extracts Claude’s `session_id`: it replaces the previous text/regex-based `grep` pipeline with a JSON-aware `jq`-based extraction that always prefers the real top-level `session_id`. The existing safety/validation behavior is kept, and the extracted value is still written atomically, preserving the hook failure exit behavior.

It also updates and expands the test suite to use realistic payloads, explicitly cover the “top-level wins over nested” edge case (and other scenarios), and skip the sandbox extraction tests when `jq` isn’t available. Related command/expectation tests were updated to match the new `jq`-guarded extraction flow, and sandbox documentation was adjusted to reflect the `jq` absence behavior.

Benefit: more reliable `session_id` capture, fewer incorrect matches from nested data, and stronger regression protection for the original issue.

```technical-details
Implementation moved to `jq -r` extraction gated by a `command -v jq` check, with UUID-shape validation and atomic sidecar writes; legacy `grep -oE`-style assertions were removed/updated accordingly.
```

### CODE REVIEW

File: src/hooks/mod.rs [hunk 1]
```diff
@@ -3521,8 +3534,35 @@ hooks_auto_accept: false
...
+            eprintln!("skipping: jq not on PATH");
+        }
+        absent
+    }
+
     #[test]
     fn test_hook_command_session_id_extracts_from_compact_payload() {
+        if skip_if_no_jq() {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply src/hooks/mod.rs:3565]: Applied at b1a1e6d1. `test_hook_command_session_id_skips_when_no_session_id` is now gated by `if skip_if_no_jq() { return; }`, matching the other 6 e2e tests that drive `run_session_id_hook`. The missing-`session_id` branch is exercised on jq-bearing runners; jq-less runners skip cleanly rather than passing through the `command -v jq` short-circuit and giving a false positive.
